app/controllers/application_controller.rb

```diff
@@ -24,68 +28,90 @@ class ApplicationController < ActionController::Base
-  end
+	def model_name(model, multiple)
+    if model.instance_of?(Class)
+    	if multiple
+				model.model_name.human(count: 2)
+ 			else	
+  			model.model_name.human
+    	end
+  	end
+	end
 
-  def self.permission
-    name.gsub('Controller', '').singularize.split('::').last.constantize.name rescue nil
-  end
+	def alert_update_resource(resource)
+		%(#{model_name(resource,false)} #{I18n.t('success_update')}.)
+	end
 
-  def current_ability
-    @current_ability ||= Ability.new(current_user)
-  end
+	def alert_create_resource(resource)
+		%(#{model_name(resource,false)} #{I18n.t('success_create')}.)
+	end
 
-  # load the permissions for the current user so that UI can be manipulated
-  def load_permissions
-    return unless current_user
-    @current_permissions = current_user.profile.posts.each do |post|
-      post.permissions.map { |i| [i.subject_class, i.action] }
-    end
-  end
+	protected
 
-  # Enables authentication and
-  def self.load_permissions_and_authorize_resource(*args)
-    load_and_authorize_resource(*args)
-    before_action(:load_permissions, *args)
-  end
+	def configure_permitted_devise_parameters
+  	devise_parameter_sanitizer.for(:sign_in) do |u|
+    	u.permit(:username, :password, :remember_me)
+  	end
+  	devise_parameter_sanitizer.for(:sign_up) do |u|
+    	u.permit(:username, :email, :password, :password_confirmation)
+  	end
+  	devise_parameter_sanitizer.for(:account_update) do |u|
+    	u.permit(:password, :password_confirmation, :current_password)
+  	end
+	end
 
-  def self.skip_authorization(*args)
-    skip_authorization_check(*args)
-    skip_before_filter(:load_permissions, *args)
-  end
+	def self.permission
+  	name.gsub('Controller', '').singularize.split('::').last.constantize.name rescue nil
+	end
 
-  def set_locale
-    locale = 'sv'
-    langs = %w{ sv en }
+	def current_ability
+  	@current_ability ||= Ability.new(current_user)
+	end
 
-    if params[:locale]
-      lang = params[:locale]
-      if langs.include? lang
-        locale = lang
-        cookies[:locale] = lang
-      end
-    else
-      if cookies[:locale]
-        lang = cookies[:locale]
-        locale = lang if langs.include? lang
-      end
-    end
-    I18n.locale = locale
-    redirect_to(:back) if params[:locale]
-  end
+	# load the permissions for the current user so that UI can be manipulated
+	def load_permissions
+  	return unless current_user
+  	@current_permissions = current_user.profile.posts.each do |post|
+    	post.permissions.map { |i| [i.subject_class, i.action] }
+  	end
+	end
 
-  def referring_action
-    Rails.application.routes.recognize_path(request.referer)[:action]
-  end
+	# Enables authentication and
+	def self.load_permissions_and_authorize_resource(*args)
+  	load_and_authorize_resource(*args)
+  	before_action(:load_permissions, *args)
+	end
+
+	def self.skip_authorization(*args)
+  	skip_authorization_check(*args)
+  	skip_before_filter(:load_permissions, *args)
+	end
+
+	def set_locale
```

Comment:
Assignment Branch Condition size for set_locale is too high. [17.8/15]<br>Inconsistent indentation detected.<br>Tab detected.